src/core/localize.rkt

```diff
@@ -28,15 +28,58 @@
 (define (batch-localize-error exprs ctx)
   (define errss
     (if (null? exprs) empty (compute-local-errors exprs ctx)))
+  (define result
   (for/list ([expr (in-list exprs)] [errs (in-list errss)])
     (sort
      (sort
       (reap [sow]
         (for ([(expr err) (in-hash errs)])
-          (unless (andmap (curry = 1) err)
-            (sow (cons err expr)))))
+          (sow (cons err expr))))
       expr<? #:key cdr)
      > #:key (compose errors-score car))))
+     result) 
+
+
+;;Returns a list of expressions sorted by increasing local cost in the form (subexpression, cost)
+(define (batch-localize-cost exprs ctx)
+  ;;Create a list of lists of subexpressions
+  (define subexprss
+    (for/list ([expr (in-list exprs)])
+      (all-subexpressions expr (context-repr ctx))))
+
+  ;Define a function to the get the cost of a expr with a repr
+  (define expr->cost  (platform-cost-proc (*active-platform*))); 
+
+  ;Create a list of (expr, cost, repr)
+  (define old-expr-cost-pair
+    (apply append
+     (map (lambda (listOfPairs)
```

Comment:
Seeing camel case in Racket is jarring 😆

Reply:
Joining these together don't make too much sense to me. You're constructing a list just to deconstruct it later.

Reply:
I join these because subexprss is a list of lists and in order to simply the expressions I just need a list of expressions to pass into simplify batch. So through processing the list I grab the list and the representation. I am basically creating two lists two parrallel lists.

Reply:
So I think you can simplify this a bit using Racket's `for*/list` construct, which works like this:
```
(for*/list ([subexprs (in-list subexprss)] [subexpr (in-list subexprs)])
  ...)
```
So what this is doing is saying that `subexprs` is in the list `subexprss` and then `subexpr` is in the nested list `subexprs`, and the whole thing returns a single list. So if you do:
```
(define subexprs
  (for*/list ([subexprs (in-list subexprss)] [subexpr (in-list subexprs)])
    (car subexpr)))
```
that will return a single list of expressions.